pkg/gatewayconfigurationserver/grpc.go

```diff
@@ -0,0 +1,100 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gatewayconfigurationserver
+
+import (
+	"context"
+	"encoding/json"
+
+	types "github.com/gogo/protobuf/types"
+	"go.thethings.network/lorawan-stack/v3/pkg/auth/rights"
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/pfconfig/cpf"
+	"go.thethings.network/lorawan-stack/v3/pkg/pfconfig/semtechudp"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+)
+
+var errInvalidConfigurationFormat = errors.DefineInvalidArgument("invalid_configuration_format", "configuration format is not supported")
+
+func (s *Server) getGateway(ctx context.Context, gtwID *ttnpb.GatewayIdentifiers) (*ttnpb.Gateway, error) {
+	cc, err := s.GetPeerConn(ctx, ttnpb.ClusterRole_ENTITY_REGISTRY, nil)
+	if err != nil {
+		return nil, err
+	}
+	client := ttnpb.NewGatewayRegistryClient(cc)
+	gtw, err := client.Get(ctx, &ttnpb.GetGatewayRequest{
+		GatewayIds: gtwID,
+		FieldMask: &types.FieldMask{
+			Paths: []string{
+				"antennas",
+				"frequency_plan_id",
+				"gateway_server_address",
+			},
+		},
+	}, s.WithClusterAuth())
+	if err != nil {
+		return nil, err
+	}
+	return gtw, nil
+}
+
+func (s *Server) GetGatewayConfiguration(ctx context.Context, req *ttnpb.GetGatewayConfigurationRequest) (*ttnpb.GetGatewayConfigurationResponse, error) {
+	if s.config.RequireAuth {
```

Comment:
I can't really think of a valid use case for disabling auth here. The configuration for a specific gateway is never public right?  We had this already in the codebase but I can't remember why. @htdvisser ?

Reply:
At some point in the past we allowed Basic Station gateways to connect without auth. We don't do that anymore on any of our production deployments, but it's not impossible that some private deployments still allow it.

Reply:
Yes but these routes are not for LBS. These are those json/toml files for UDP gateways. So now anyone can just insert a gateway ID here and get the configuration for a particular gateway. 
Not that it's a huge security issue but I think we can avoid some confusion here.